card/src/main/java/com/adyen/checkout/card/CardViewNew.kt

```diff
@@ -0,0 +1,750 @@
+/*
+ * Copyright (c) 2022 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by josephj on 16/9/2022.
+ */
+package com.adyen.checkout.card
+
+import android.app.Activity
+import android.content.Context
+import android.content.ContextWrapper
+import android.text.Editable
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.View
+import android.view.View.OnFocusChangeListener
+import android.view.WindowManager
+import android.widget.AdapterView
+import android.widget.LinearLayout
+import androidx.annotation.StringRes
+import androidx.core.view.isVisible
+import com.adyen.checkout.card.data.CardType
+import com.adyen.checkout.card.data.DetectedCardType
+import com.adyen.checkout.card.data.ExpiryDate
+import com.adyen.checkout.card.databinding.CardViewNewBinding
+import com.adyen.checkout.card.ui.SecurityCodeInput
+import com.adyen.checkout.card.ui.model.AddressListItem
+import com.adyen.checkout.card.util.InstallmentUtils
+import com.adyen.checkout.components.api.ImageLoader
+import com.adyen.checkout.components.base.ComponentDelegate
+import com.adyen.checkout.components.extensions.isVisible
+import com.adyen.checkout.components.extensions.setLocalizedHintFromStyle
+import com.adyen.checkout.components.extensions.setLocalizedTextFromStyle
+import com.adyen.checkout.components.ui.ComponentMode
+import com.adyen.checkout.components.ui.ComponentViewNew
+import com.adyen.checkout.components.ui.FieldState
+import com.adyen.checkout.components.ui.Validation
+import com.adyen.checkout.components.ui.view.AdyenTextInputEditText
+import com.adyen.checkout.components.ui.view.RoundCornerImageView
+import com.adyen.checkout.core.exception.CheckoutException
+import com.adyen.checkout.core.util.BuildUtils
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+/**
+ * CardView for [CardComponent].
+ */
+@Suppress("TooManyFunctions", "LargeClass")
+class CardViewNew @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) :
+    LinearLayout(
+        context,
+        attrs,
+        defStyleAttr
+    ),
+    ComponentViewNew {
+
+    companion object {
```

Comment:
I'd suggest to follow to [official Kotlin conventions for a class layout](https://kotlinlang.org/docs/coding-conventions.html#class-layout):
> The contents of a class should go in the following order:
> 1. Property declarations and initializer blocks
> 2. Secondary constructors
> 3. Method declarations
> 4. Companion object